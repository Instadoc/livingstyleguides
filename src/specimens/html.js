module.exports = (userOptions = {}) => {
    const defaultOptions = {
        collapsedByDefault: [],
        expandedByDefault: ["html", "css", "js"],
    }
    const options = Object.assign({}, defaultOptions, userOptions)

    return {
        name: "html",
        options,

        test: (specimen, options, config) => specimen.blocks[0].type === "html",

        render: (specimen, options, config) =>
            specimen.blocks
                .filter((block) => block.type === "html")
                .map((block) => block.content)
                .join("\n"),

        initDom: (specimen, shadowRoot, options, config) => {
            const inlineCss = specimen.blocks
                .filter((block) => block.type === "css")
                .map((block) => block.content)
                .join("\n")

            const inlineJs = specimen.blocks
                .filter((block) => block.type === "js")
                .map((block) => block.content)
                .join("\n")

            if (inlineCss) {
                const style = document.createElement("style")
                style.textContent = inlineCss
                shadowRoot.appendChild(style)
            }

            if (inlineJs) {
                const specimenId = shadowRoot.host.getAttribute("id")
                const script = document.createElement("script")
                script.textContent = `(function() {
                    const document = window.document.getElementById("${specimenId}");
                    ${inlineJs}
                })()`
                setTimeout(() => {
                    shadowRoot.appendChild(script)
                })
            }

            const optionsNode = document.createTextNode(`Options: ${JSON.stringify(options)}`)
            shadowRoot.appendChild(optionsNode)
        },
    }
}
