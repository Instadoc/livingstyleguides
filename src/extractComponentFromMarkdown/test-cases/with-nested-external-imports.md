---
name: Component Name
category: Component Category
---

This is the first specimen:

```specimen-1.html
<b>Specimen 1</b>
```

```specimen-1.css
b { color: red }
```

This is a specimen with nested external imports:

```specimen-2.html
import './with-nested-external-imports.external-import.html'
<b>Specimen 2</b>
```

```specimen-2.js
import './with-nested-external-imports.external-import.js'
import './with-nested-external-imports.external-import-2.js'
var specimen = 2
```

```specimen-2.css
b { color: green }
```
