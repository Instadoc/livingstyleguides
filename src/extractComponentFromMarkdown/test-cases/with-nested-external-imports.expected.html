<p>This is the first specimen:</p>
<pre><code class="language-specimen-1.html">&#x3C;b>Specimen 1&#x3C;/b>
</code></pre>
<pre><code class="language-specimen-1.css">b { color: red }
</code></pre>
<p>This is a specimen with nested external imports:</p>
<pre><code class="language-specimen-2.css">span { color: blue }
</code></pre>
<pre><code class="language-specimen-2.html">&#x3C;span>Specimen 2 external import&#x3C;/span>
</code></pre>
<pre><code class="language-specimen-2.html">&#x3C;b>Specimen 2&#x3C;/b>
</code></pre>
<pre><code class="language-specimen-2.js">var externalImport1 = 'one'
</code></pre>
<pre><code class="language-specimen-2.js">var externalImport3 = 'three'
var externalImport4 = 'four'
var externalImport2 = 'two'
</code></pre>
<pre><code class="language-specimen-2.css">b { color: green }
</code></pre>